src/toil/lib/threading.py

```diff
@@ -92,9 +89,15 @@ def tryRun( self ):
     def join( self, *args, **kwargs ):
         super( ExceptionalThread, self ).join( *args, **kwargs )
         if not self.is_alive( ) and self.exc_info is not None:
-            type, value, traceback = self.exc_info
+            exc_type, exc_value, traceback = self.exc_info
             self.exc_info = None
-            raise_(type, value, traceback)
+            if exc_value is not None:
+                exc = exc_type(exc_value)
+            else:
+                exc = exc_type
+            if exc.__traceback__ is not traceback:
+                raise exc.with_traceback(traceback)
+            raise exc
```

Comment:
I've seen this incantation twice now (in the `exceptions.py` file too).  Consider making these into one function.

Reply:
Done, thanks!